deepchem/molnet/load_function/uspto_datasets.py

```diff
@@ -4,93 +4,117 @@
-  # so there is no "tasks" field.
-  uspto_tasks = []
-  if split is not None:
-    raise ValueError("Train/valid/test not yet supported.")
-  # Download USPTO dataset
-  if reload:
-    save_folder = os.path.join(save_dir, "uspto-featurized", str(featurizer))
-    if featurizer == "smiles2img":
-      img_spec = kwargs.get("img_spec", "std")
-      save_folder = os.path.join(save_folder, img_spec)
-    save_folder = os.path.join(save_folder, str(split))
-
-    loaded, all_dataset, transformers = deepchem.utils.data_utils.load_dataset_from_disk(
-        save_folder)
-    if loaded:
-      return uspto_tasks, all_dataset, transformers
-
-  dataset_file = os.path.join(data_dir,
-                              "2008-2011_USPTO_reactionSmiles_filtered.zip")
-  if not os.path.exists(dataset_file):
-    deepchem.utils.data_utils.download_url(url=USPTO_URL, dest_dir=data_dir)
-
-  # Unzip
-  unzip_dir = os.path.join(data_dir, "2008-2011_USPTO_reactionSmiles_filtered")
-  if not os.path.exists(unzip_dir):
-    deepchem.utils.data_utils.unzip_file(dataset_file, dest_dir=unzip_dir)
-  # Unzipped file is a tap seperated values file (despite the .txt)
-  filename = os.path.join(unzip_dir,
-                          "2008-2011_USPTO_reactionSmiles_filtered.txt")
-  rxns = []
-  from rdkit.Chem import rdChemReactions
-  with open(filename) as tsvfile:
-    reader = csv.reader(tsvfile, delimiter="\t")
-    for ind, row in enumerate(reader):
-      if ind > num_to_load:
-        break
-      if verbose:
-        print("Loading reaction %d" % ind)
-      # The first element in the row is the reaction smarts
-      smarts = row[0]
-      # Sometimes smarts have extraneous information at end of form "
-      # |f:0" that causes parsing to fail. Not sure what this information
-      # is, but just ignoring for now.
-      smarts = smarts.split(" ")[0]
-      rxn = rdChemReactions.ReactionFromSmarts(smarts)
-      rxns.append(rxn)
-  rxn_array = np.array(rxns)
-  # Make up dummy labels since DiskDataset.from_numpy doesn't allow
-  # creation from just features for now.
-  y = np.ones(len(rxn_array))
-  # TODO: This dataset isn't saved to disk so reload doesn't happen.
-  rxn_dataset = DiskDataset.from_numpy(rxn_array, y)
-  transformers = []
-  return uspto_tasks, (rxn_dataset, None, None), transformers
+
+USPTO_MIT_TRAIN = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_MIT_train.csv"
+USPTO_MIT_TEST = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_MIT_test.csv"
+USPTO_MIT_VALID = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_MIT_val.csv"
+
+USPTO_STEREO_TRAIN = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_STEREO_train.csv"
+USPTO_STEREO_TEST = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_STEREO_test.csv"
+USPTO_STEREO_VALID = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_STEREO_val.csv"
```

Comment:
Is it possible to combine these into a single dataset with an additional label that specifies for each point whether it is in train/val/test? That way you can load a single dataset (like `_MolnetLoader`) expects, and do a sort of dummy split that constructs train/val/test.

Reply:
It seems a bit hacky, but I think it should be possible. Do you think I should try writing a custom splitter or just go for this?

Reply:
You could have a toggle in `_MolnetLoader` that goes something as `precomputed_split=False`, and extend it to handle those cases where a precomputed split exists. In this case, you can also decide whether to override `splitter` or not, if it is not `None`.